sqlglot/generator.py

```diff
@@ -564,21 +571,18 @@ def create_sql(self, expression: exp.Create) -> str:
                     if index.args.get("columns")
                     else ""
                 )
-                if index.args.get("primary") and properties_locs.get(
-                    exp.Properties.Location.POST_INDEX
-                ):
-                    postindex_props_sql = self.properties(
-                        exp.Properties(
-                            expressions=properties_locs[exp.Properties.Location.POST_INDEX]
-                        ),
-                        wrapped=False,
-                    )
-                    ind_columns = f"{ind_columns} {postindex_props_sql}"
+                ind_sql = f"{ind_unique}{ind_primary}{ind_amp} INDEX{ind_name}{ind_columns}"
 
-                indexes_sql.append(
-                    f"{ind_unique}{ind_primary}{ind_amp} INDEX{ind_name}{ind_columns}"
-                )
+                if len(indexes_sql) == 0:
+                    if index.args.get("primary"):
+                        indexes_sql.append(f"{ind_sql}{postindex_props_sql}")
+                    else:
+                        indexes_sql.append(f"{postindex_props_sql}{ind_sql}")
+                else:
+                    indexes_sql.append(ind_sql)
```

Comment:
What about checking `if indexes_sql` first to avoid `len` altogether? Seems fine either way, although ideally we'd like to simplify this a bit, if possible.